ocaml-lsp-server/src/workspace_symbol.ml

```diff
@@ -172,14 +172,14 @@ let run ({ query; _ } : WorkspaceSymbolParams.t) (rootUri : Uri.t option) =
     match query with
     | "" -> Ok all_symbols
     | query ->
-      let reg_exp = Str.regexp_string query in
+      let reg_exp = Re.Str.regexp_string query in
       let symbols =
         all_symbols
         |> List.filter ~f:(fun symbol ->
                try
                  let () =
                    ignore
-                     (Str.search_forward reg_exp
+                     (Re.Str.search_forward reg_exp
```

Comment:
I believe `Re.Str` suffers from the same problems. You can just use the vanilla `Re` api instead.